diff --git a/capacitor/src/main/java/com/getcapacitor/plugin/SystemBars.java b/capacitor/src/main/java/com/getcapacitor/plugin/SystemBars.java
index c84923ca8bdd55cbe99d1c4f0ee5c6b17fec98ac..1a45ec2a4d502e0bda580426baca385272924a1c 100644
--- a/capacitor/src/main/java/com/getcapacitor/plugin/SystemBars.java
+++ b/capacitor/src/main/java/com/getcapacitor/plugin/SystemBars.java
@@ -56,7 +56,8 @@ public class SystemBars extends Plugin {
     private boolean insetHandlingEnabled = true;
     private boolean hasViewportCover = false;
 
-    private String currentStyle = STYLE_DEFAULT;
+    private String currentStatusBarStyle = STYLE_DEFAULT;
+    private String currentGestureBarStyle = STYLE_DEFAULT;
 
     @Override
     public void load() {
@@ -85,10 +86,13 @@ public class SystemBars extends Plugin {
     protected void handleOnConfigurationChanged(Configuration newConfig) {
         super.handleOnConfigurationChanged(newConfig);
 
-        setStyle(currentStyle, "");
+        setStyle(currentGestureBarStyle, BAR_GESTURE_BAR);
+        setStyle(currentStatusBarStyle, BAR_STATUS_BAR);
     }
 
     private void initSystemBars() {
+        WindowCompat.setDecorFitsSystemWindows(this.getActivity().getWindow(), false);
+
         String style = getConfig().getString("style", STYLE_DEFAULT).toUpperCase(Locale.US);
         boolean hidden = getConfig().getBoolean("hidden", false);
 
@@ -162,13 +166,11 @@ public class SystemBars extends Plugin {
     }
 
     private void initSafeAreaCSSVariables() {
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.VANILLA_ICE_CREAM && insetHandlingEnabled) {
-            View v = (View) this.getBridge().getWebView().getParent();
-            WindowInsetsCompat insets = ViewCompat.getRootWindowInsets(v);
-            if (insets != null) {
-                Insets safeAreaInsets = calcSafeAreaInsets(insets);
-                injectSafeAreaCSS(safeAreaInsets.top, safeAreaInsets.right, safeAreaInsets.bottom, safeAreaInsets.left);
-            }
+        View v = (View) this.getBridge().getWebView().getParent();
+        WindowInsetsCompat insets = ViewCompat.getRootWindowInsets(v);
+        if (insets != null) {
+            Insets safeAreaInsets = calcSafeAreaInsets(insets);
+            injectSafeAreaCSS(safeAreaInsets.top, safeAreaInsets.right, safeAreaInsets.bottom, safeAreaInsets.left);
         }
     }
 
@@ -184,10 +186,8 @@ public class SystemBars extends Plugin {
                 // We need to correct for a possible shown IME
                 v.setPadding(0, 0, 0, keyboardVisible ? imeInsets.bottom : 0);
 
-                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.VANILLA_ICE_CREAM && hasViewportCover && insetHandlingEnabled) {
-                    Insets safeAreaInsets = calcSafeAreaInsets(insets);
-                    injectSafeAreaCSS(safeAreaInsets.top, safeAreaInsets.right, safeAreaInsets.bottom, safeAreaInsets.left);
-                }
+                Insets safeAreaInsets = calcSafeAreaInsets(insets);
+                injectSafeAreaCSS(safeAreaInsets.top, safeAreaInsets.right, safeAreaInsets.bottom, safeAreaInsets.left);
 
                 return new WindowInsetsCompat.Builder(insets)
                     .setInsets(
@@ -217,10 +217,8 @@ public class SystemBars extends Plugin {
                 .setInsets(WindowInsetsCompat.Type.systemBars() | WindowInsetsCompat.Type.displayCutout(), Insets.of(0, 0, 0, 0))
                 .build();
 
-            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.VANILLA_ICE_CREAM && hasViewportCover && insetHandlingEnabled) {
-                Insets safeAreaInsets = calcSafeAreaInsets(newInsets);
-                injectSafeAreaCSS(safeAreaInsets.top, safeAreaInsets.right, safeAreaInsets.bottom, safeAreaInsets.left);
-            }
+            Insets safeAreaInsets = calcSafeAreaInsets(newInsets);
+            injectSafeAreaCSS(safeAreaInsets.top, safeAreaInsets.right, safeAreaInsets.bottom, safeAreaInsets.left);
 
             return newInsets;
         });
@@ -259,8 +257,6 @@ public class SystemBars extends Plugin {
     }
 
     private void setStyle(String style, String bar) {
-        currentStyle = style;
-
         if (style.equals(STYLE_DEFAULT)) {
             style = getStyleForTheme();
         }
@@ -268,10 +264,12 @@ public class SystemBars extends Plugin {
         Window window = getActivity().getWindow();
         WindowInsetsControllerCompat windowInsetsControllerCompat = WindowCompat.getInsetsController(window, window.getDecorView());
         if (bar.isEmpty() || bar.equals(BAR_STATUS_BAR)) {
+            currentStatusBarStyle = style;
             windowInsetsControllerCompat.setAppearanceLightStatusBars(!style.equals(STYLE_DARK));
         }
 
         if (bar.isEmpty() || bar.equals(BAR_GESTURE_BAR)) {
+            currentGestureBarStyle = style;
             windowInsetsControllerCompat.setAppearanceLightNavigationBars(!style.equals(STYLE_DARK));
         }
 
